util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,264 @@
+            num = tryDoubleParse(line, doublePredicate);
+            if (num.isEmpty()) {
+                System.out.println("Invalid input.");
+            }
+        }
+        return num.getAsDouble();
+    }
+
+    /**
+     * Tries to parse a Double from the string and validate it.
+     *
+     * @param line            the string to parse.
+     * @param doublePredicate the predicate used to validate the double.
+     * @return a validated Double or nothing.
+     */
+    private OptionalDouble tryDoubleParse(String line, Predicate<? super Double> doublePredicate) {
+        try {
+            var num = Double.parseDouble(line);
+            return doublePredicate.test(num) ? OptionalDouble.of(num) : OptionalDouble.empty();
+        }
+        catch (NumberFormatException e) {
+            return OptionalDouble.empty();
+        }
+    }
+
+    /**
+     * Gets the next yes or no answer.
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     *
+     * @param prompt the prompt to display to the user.
+     * @return true for yes and false for no.
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public boolean nextYesNo(String prompt) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).trim().toLowerCase();
+        return YES_SYNONYMS.contains(result);
+    }
+
+    /**
+     * Waits for the user to press &lt;enter&gt;.
+     * The prompt defaults to "Press when ready".
+     * Calls {@link InputUtility#pause(String)}.
+     *
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public void pause() throws IOException {
+        pause("Press enter when ready");
+    }
+
+    /**
+     * Uses prompt entered and waits for the user to press &lt;enter&gt;.
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     *
+     * @param prompt the prompt to display to the user.
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public void pause(String prompt) throws IOException {
+        this.nextString(prompt, String::isBlank);
+    }
+
+    /**
+     * Get a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     *
+     * @return a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     */
+    public static Predicate<String> yesOrNo() {
+        return s -> {
+            var cleanedString = s.trim().toLowerCase();
+            return YES_SYNONYMS.contains(cleanedString) || NO_SYNONYMS.contains(cleanedString);
+        };
+    }
+
+
+    /**
+     * Get a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     *
+     * @param firstOption the first of the strings to match to.
+     * @param otherOptions the strings for the input to match to.
```

Comment:
In all methods in this class that have a varargs parameter, please rewrite the description so that it refers to "the other strings" instead of just "the strings".